### ISSUE TITLE
Node-Attention in SemiGNN

### ISSUE THREAD

**Issue by wangtsiao** [safe-graph/DGFraud#36]
您好！再次仔细拜读了SemiGNN的实现代码，我存在一个疑问。

- 代码中有关Node-Attention机制的实现，见[代码链接](https://github.com/safe-graph/DGFraud/blob/8d43fd23d700d7f176dc199e624bd5a1bb945ef6/base_models/layers.py#L200)，其中H_v是下式中的注意力参数，代码中H_v的shape是[hidden_size, 1]，其中hidden_size是节点嵌入向量的长度，这里按照论文理解的话，H_v的shape应该是由当前节点的邻居节点数量所决定，但这样H_v的shape不是固定的，网络就没办法back propagation训练。
![image](https://user-images.githubusercontent.com/70847926/101356529-2c743400-38d3-11eb-9a48-fec86c5d92a1.png)

感谢您花费宝贵的时间，谢谢！

**Comment by YingtongDou** [comment]
The attention layer H_v has a fixed shape to encode the input embedding of every node. During the backpropagation, only the parameters of H_v are updated which is independent of the graph structure.

[This line](https://github.com/safe-graph/DGFraud/blob/8d43fd23d700d7f176dc199e624bd5a1bb945ef6/base_models/layers.py#L216) will compute the softmax attention coefficient shown in the equation above.

The [GAT](https://arxiv.org/abs/1710.10903) paper and its [implementation](https://github.com/PetarV-/GAT) will help you understand how the attention layer works.
